### PR TITLE
Fix python lints, make python lint ci conditional

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* related to #3033

--

* Turned out most of the python lints issues I had locally (and did not see on ci) are about opencv-python missmatches. The way I fixed it should work fine with either though now!
* Various python lints on ci now use pixi and run only when py files change.
* update pixi version used on ci

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [rerun.io/viewer](https://rerun.io/viewer/pr/5897)
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/5897?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/5897?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5897)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by Wumpf** [comment]
(do not merge since this goes into another not yet merged pr)

File: examples/python/face_tracking/main.py [hunk 1]
```diff
@@ -302,7 +302,7 @@ def download_file(url: str, path: Path) -> None:
             f.write(chunk)
 
 
-def resize_image(image: npt.NDArray[np.uint8], max_dim: int | None) -> npt.NDArray[np.uint8]:
+def resize_image(image: npt.NDArray[Any], max_dim: int | None) -> npt.NDArray[Any]:
```
Comment (Wumpf) [inline examples/python/face_tracking/main.py:305]: wrong. should also be cv2.typing.MatLike

File: .github/workflows/on_pull_request.yml [hunk 2]
```diff
@@ -55,6 +55,21 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.ref || '' }}
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            python_changes:
+              - '**/*.py'
```
Comment (emilk) [inline .github/workflows/on_pull_request.yml:71]: Should probably also trigger if a `requirements.txt` or `pyproject.toml` file changes

**Review by emilk — APPROVED** [review]

File: .github/workflows/on_pull_request.yml (hunk 2 above)
Comment (Wumpf) [reply .github/workflows/on_pull_request.yml:71]: 👍

**Review by emilk — APPROVED** [review]

**Pull request merged**